### PR TITLE
store/mongo: handle mongodb client creation error

### PR DESCRIPTION
**Pull request by kjaskiewiczz**
Issues: https://tracker.mender.io/browse/MEN-2971

### CODE REVIEW

File: store/mongo/datastore_mongo.go [hunk 1]
```diff
@@ -121,6 +121,9 @@ func NewDataStoreMongo(config DataStoreMongoConfig) (*DataStoreMongo, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
 	defer cancel()
 	c, err := mongo.Connect(ctx, clientOptions)
+	if err != nil {
+		return nil, err
+	}
```
Comment (merlin-northern) [inline store/mongo/datastore_mongo.go]: can we add Errorf log here?

File: tests/common.py [hunk 2]
```diff
@@ -55,7 +55,7 @@ def mongo():
 
 def mongo_cleanup(mongo):
     dbs = mongo.database_names()
-    dbs = [d for d in dbs if d not in ['local', 'admin']]
+    dbs = [d for d in dbs if d not in ['local', 'admin', 'config']]
```
Comment (merlin-northern) [inline tests/common.py]: after upgrade to v3.6 do we need to change it for every service?

Comment (kjaskiewiczz) [reply tests/common.py]: yes

File: store/mongo/datastore_mongo.go (hunk 1 above)
Comment (kjaskiewiczz) [reply store/mongo/datastore_mongo.go]: why? how it will be different from what we have right now?

Comment (tranchitella) [reply store/mongo/datastore_mongo.go]: Looks okay to me.

File: tests/common.py (hunk 2 above)
Comment (tranchitella) [reply tests/common.py]: I confirm this change is just fine and needs to be applied to a similar code across all the codebase.

**Pull request merged**